docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Clustering/KMeans.cs

```diff
@@ -1,51 +1,114 @@
-                .Append(ml.Clustering.Trainers.KMeans(outputColumnName, numberOfClusters: 2));
+            // Convert IDataView object to a list.
+            var predictions = mlContext.Data.CreateEnumerable<Prediction>(transformedTestData, reuseRowObject: false).ToList();
 
-            var model = pipeline.Fit(trainData);
+            // Look at 5 predictions
+            foreach (var p in predictions.Take(2))
+                Console.WriteLine($"Label: {p.Label}, Prediction: {p.PredictedLabel}");
+            foreach (var p in predictions.TakeLast(3))
+                Console.WriteLine($"Label: {p.Label}, Prediction: {p.PredictedLabel}");
+
+            // Expected output:
+            //   Label: 1, Prediction: 1
+            //   Label: 1, Prediction: 1
+            //   Label: 2, Prediction: 2
+            //   Label: 2, Prediction: 2
+            //   Label: 2, Prediction: 2
+
+            // Evaluate the overall metrics
+            var metrics = mlContext.Clustering.Evaluate(transformedTestData, "Label", "Score", "Features");
+            Console.WriteLine($"Normalized Mutual Information: {metrics.NormalizedMutualInformation:F2}");
+            Console.WriteLine($"Average Distance: {metrics.AverageDistance:F2}");
+            Console.WriteLine($"Davies Bouldin Index: {metrics.DaviesBouldinIndex:F2}");
+            
+            // Expected output:
+            //   Normalized Mutual Information: 0.95
+            //   Average Distance: 4.17
+            //   Davies Bouldin Index: 2.87
 
             // Get cluster centroids and the number of clusters k from KMeansModelParameters.
             VBuffer<float>[] centroids = default;
-            int k;
 
-            var modelParams = model.LastTransformer.Model;
-            modelParams.GetClusterCentroids(ref centroids, out k);
+            var modelParams = model.Model;
+            modelParams.GetClusterCentroids(ref centroids, out int k);
+            Console.WriteLine($"The first 3 coordinates of the first centroid are: ({string.Join(", ", centroids[0].GetValues().ToArray().Take(3))})");
+            Console.WriteLine($"The first 3 coordinates of the second centroid are: ({string.Join(", ", centroids[1].GetValues().ToArray().Take(3))})");
 
-            var centroid = centroids[0].GetValues();
-            Console.WriteLine($"The coordinates of centroid 0 are: ({string.Join(", ", centroid.ToArray())})");
-
-            //  Expected output similar to:
-            //      The coordinates of centroid 0 are: (26, 6, 1)
+            // Expected output similar to:
+            //   The first 3 coordinates of the first centroid are: (0.6035213, 0.6017533, 0.5964218)
+            //   The first 3 coordinates of the second centroid are: (0.4031044, 0.4175443, 0.4082336)
             //
             // Note: use the advanced options constructor to set the number of threads to 1 for a deterministic behavior.
```

Comment:
>// Note: use the advanced options constructor to set the number of threads to 1 for a deterministic behavior. [](start = 12, length = 109)

This comment seems out of context. Specifically what does advance options constructor mean here?